lib/rules/color-no-invalid-hex/index.js

```diff
@@ -23,36 +23,39 @@ const rule = function(actual) {
     root.walkDecls(decl => {
       const declString = decl.toString();
 
-      styleSearch({ source: declString, target: "#" }, match => {
-        // If there's not a colon, comma, or whitespace character before, we'll assume this is
-        // not intended to be a hex color, but is instead something like the
-        // hash in a url() argument
-        if (!/[:,\s]/.test(declString[match.startIndex - 1])) {
-          return;
-        }
+      styleSearch(
+        { source: declString, target: "#", strings: "check" },
```

Comment:
I suspect the way to fix this is to migrate the rule away from `styleSearch` and to `postcss-value-parser`. There wasn't a value parser around when this rule was created, but now we have one and it is proving a more robust way of dealing with the values of declarations.

There an example of the value parser in use in the [color-named rule](https://github.com/stylelint/stylelint/blob/348c139ef1b1c69237a71225d55dfdee73cce39c/lib/rules/color-named/index.js#L73).

e.g.

```js
root.walkDecls(decl => {
  valueParser(decl.value).walk(({value, type, sourceIndex) => {
    if (type !== "word") return;
    const hexMatch = /^#[0-9A-Za-z]+/.exec(value);
    ..
  }
}
```